src/System.Memory/tests/ReadOnlyMemory/Equality.cs

```diff
@@ -84,6 +84,58 @@ public static void EmptyMemoryNotUnified()
             Assert.True(right.Equals(left));
         }
 
+        [Fact]
+        public static void EqualityThroughInterface_True()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<ReadOnlyMemory<int>> left = new ReadOnlyMemory<int>(array, 2, 3);
+            IEquatable<ReadOnlyMemory<int>> right = new ReadOnlyMemory<int>(array, 2, 3);
+
+            Assert.True(left.Equals(right));
+            Assert.True(right.Equals(left));
+        }
+
+        [Fact]
+        public static void EqualityThroughInterface_Reflexivity()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<ReadOnlyMemory<int>> left = new ReadOnlyMemory<int>(array, 2, 3);
+
+            Assert.True(left.Equals(left));
+        }
+
+        [Fact]
+        public static void EqualityThroughInterface_IncludesLength()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<ReadOnlyMemory<int>> left = new ReadOnlyMemory<int>(array, 2, 3);
+            IEquatable<ReadOnlyMemory<int>> right = new ReadOnlyMemory<int>(array, 2, 2);
+
+            Assert.False(left.Equals(right));
+            Assert.False(right.Equals(left));
+        }
+
+        [Fact]
+        public static void EqualityThroughInterface_IncludesBase()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<ReadOnlyMemory<int>> left = new ReadOnlyMemory<int>(array, 1, 3);
```

Comment:
How about mismatched? Where one is `IEquatable` and the other is a regular `Memory<int>`?

Reply:
Sure, why not.

Reply:
@ahsonkhan Please have a look at the review comments I left below. I think the tests for mismatched types should go in `EqualityThroughInterface_True` case.